In [1]:
import json
import os
path = '/home/drrndrrnprn/nlp/ABST/datasets/semeval-pengb/analyzed/concat.json'
json_file = []
with open(path, 'r') as f:
    json_file += json.load(f)
print(len(json_file))

3110


In [5]:
aos_path = "/home/drrndrrnprn/nlp/ABST/datasets/semeval-pengb/analyzed/test_asp.txt"
raw_aos_list = []
with open(aos_path, "r", encoding="utf-8") as f:
    for aos_line in f:
        new_aos_line = [aos.split(',') for aos in aos_line.strip("\n").split('\t')]
        raw_aos_list.append(new_aos_line)
size_text = len(raw_aos_list)
size_aos = [len(line) for line in raw_aos_list]

dic_size_aos = {}
for s in range(1,max(size_aos)+1):
    list_idx_aos = []
    for i, size_l_aos in enumerate(size_aos):
        if size_l_aos == s:
            list_idx_aos.append(i)
    dic_size_aos[s] = list_idx_aos
count_aoses = [len(s) for s in dic_size_aos.values()]

all_size_aos = sum(size_aos)
print(f"text size: {size_text},\n triplet size: {all_size_aos},\n triplet size by ts: {count_aoses}")
count_aos_path = os.path.dirname(aos_path) + '/test_e_index.json'
with open(count_aos_path, "w") as f:
    json.dump(dic_size_aos, f, indent = 4)

text size: 600,
 triplet size: 1080,
 triplet size by ts: [292, 198, 69, 27, 10, 3, 0, 0, 1]


concat = 3110
train = text size: 1910, triplet size: 3431, [973, 576, 223, 92, 27, 13, 4, 0, 0, 0, 0, 0, 0, 2]
dev = text size: 600, triplet size: 1055, [326, 165, 58, 36, 10, 4, 1]
test = text size: 600, triplet size: 1080, {1:292, 2:198, 3:69, 4:27, 5:10, 6:3, 7:0, 8:0, 9:1}
